src/Audio.md

```diff
@@ -0,0 +1,110 @@
+
+::: warning
+
+The Game Boy's APU is actually full of tricky details; this chapter will mostly describe the intended / common behavior, and often paper over bugs & quirks.
+Readers wishing to learn more should read the [APU details](<#Audio Details>) chapter.
+
+:::
+
+### Triggering
+
+**Triggering** a channel causes it to turn on if it wasn't[^trig_dac_off], and to start playing its wave from the beginning[^pulse_restart].
+Most changes to a channel's parameters take effect immediately, but some require re-triggering the channel.
+
+### Volume & envelope
+
+The volume can be controlled in two ways: there is a "master volume" control[^vol_knob] (which has separate settings for the left and right outputs), and each channel's volume can be individually set as well (CH3's less precisely than the others).
+
+Additionally, an [**envelope**](https://en.wikipedia.org/wiki/Envelope_(music)) can be configured for all channels except CH3, which allows automatically adjusting the volume over time.
+The parameters that can be controlled are how many times the envelope will be stepped, in which direction, and how long between steps.
+All envelopes are ticked at 64 Hz, and every 1–7 of those ticks, the volume will be increased or decreased.
+
+### Length timer
+
+All channels can be individually set to automatically shut themselves down after a certain amount of time.
+
+If the functionality is enabled, a channel's **length timer** ticks up[^len_cnt_dir] at 256 Hz (tied to [DIV-APU](<#DIV-APU>)) from the value it's initially set at.
+When the length timer reaches 64, the channel is turned off.
+
+### Frequency
+
+Music notes and audio waves are typically manipulated in terms of **frequency**[^pitch], i.e. how often the signal repeats per second.
+However, as explained above, the Game Boy APU primarily works with durations; thus, **wavelengths** will be used instead of frequency.[^len_raw]
+
+::: warning
+
+The term "wavelength" throughout this document does *not* designate the inverse of the frequency, but instead a quantity akin to it.
+See the description of each NRx3 register for more information.
+
+:::
+
+---
+
+[^trig_dac_off]:
+If [the channel's DAC](#DACs) is off, the channel will not turn on.
+
+[^pulse_restart]:
+Except for pulse channels, whose index into their waves is only ever reset by turning the APU off.
+This is not noticeable when playing sound, and only matters for "advanced" APU usage.
+
+[^vol_knob]:
+This is separate from the physical volume knob located on the side of the console.
+
+[^len_cnt_dir]:
+Internally, the length timer is inverted when written, and *that* ticks down until it reaches 0.
+But the effect is as if the counter ticked up.
+
+[^pitch]:
+There is also **pitch**, which is merely a measure of how we perceive frequency.
+The higher the frequency, the higher the pitch; therefore, pitch will be omitted from the rest of the document.
```

Comment:
Actually, rather than omitting pitch from the document, maybe using pitch as the preferred term instead of frequency is useful. Consider especially the noise channel, which for some settings doesn't have a clearly defined frequency, in terms of what the human ear can perceive. (For example 15-bit polynomial setting will produce a cycle that can be more than a second long.) Yet, it would be correct to say that we can vary the pitch of the sound.

Reply:
That doesn't mesh with the "registers" document, which expresses frequencies in terms of what is written to the registers. Pitch can be derived from that, but it's non-trivial.